### PR TITLE
Remove inconsistency with readOnly and isEditable

### PR DESCRIPTION
**Pull request by ranquild**
Now `isEditable() === !readOnly()` in all cases. Previously there was an edge case with sandboxing where `isEditable()` was `false` and `readOnly` was `false`, which is incorrect. `isEditable` is more strict.

### CODE REVIEW

File: e2e/test/scenarios/visualizations-tabular/pivot_tables.cy.spec.js [hunk 1]
```diff
@@ -26,8 +26,6 @@ const {
   PEOPLE,
   REVIEWS,
   REVIEWS_ID,
-  ANALYTIC_EVENTS,
```
Comment (ranquild) [inline e2e/test/scenarios/visualizations-tabular/pivot_tables.cy.spec.js:29]: New sample DB tables shouldn't be used in e2e tests

File: frontend/src/metabase-lib/queries/NativeQuery.ts [hunk 2]
```diff
@@ -154,15 +154,15 @@ export default class NativeQuery extends AtomicQuery {
   /**
    * Returns true if the database metadata (or lack thererof indicates the user can modify and run this query
    */
-  readOnly() {
-    const database = this.database();
-    return !database || database.native_permissions !== "write";
+  readOnly(): boolean {
```
Comment (oleggromov) [inline frontend/src/metabase-lib/queries/NativeQuery.ts:157]: I am guessing the comment should be updated too?

Comment (ranquild) [reply frontend/src/metabase-lib/queries/NativeQuery.ts:157]: Fixed

Comment (kulyk) [reply frontend/src/metabase-lib/queries/NativeQuery.ts:157]: fwiw, i don't think we really need that comment anymore, it's clear it's the opposite of `isEditable` from the method itself

**Review by kulyk — APPROVED** [review]
